src/services/formatting/rules.ts

```diff
@@ -321,6 +321,10 @@ namespace ts.formatting {
             rule("NoSpaceAfterCloseBracket", SyntaxKind.CloseBracketToken, anyToken, [isNonJsxSameLineTokenContext, isNotBeforeBlockInFunctionDeclarationContext], RuleAction.Delete),
             rule("SpaceAfterSemicolon", SyntaxKind.SemicolonToken, anyToken, [isNonJsxSameLineTokenContext], RuleAction.Space),
 
+            // Remove extra space between for and await
+            rule("SpaceAfterFor", SyntaxKind.ForKeyword, anyToken, [ isNonJsxSameLineTokenContext ], RuleAction.Space),

```

Comment:
call this `SpaceBetweenForAndAwaitKeyword`. also make it between `SyntaxKind.ForKeyword` and `SyntaxKind.AwaitKeyword`, the others are already covered by other rules.